sklearn/impute/_iterative.py

```diff
@@ -581,6 +582,36 @@ def fit_transform(self, X, y=None):
             self.n_iter_ = 0
             return super()._concatenate_indicator(Xt, X_indicator)
 
+        # Convert min and max values to float arrays if array-like
+        # Else (i.e, if scalar) broadcast it to an array of shape=n_features
+        self._min_value = (
+            as_float_array(self.min_value, force_all_finite=False)
```

Comment:
Should we just use numpy.broadcast_to?

Reply:
`numpy.broadcast_to` is nifty, but it misses one critical condition - when `min_value` or `max_value` is `None`. `broadcast_to` converts `None` to an object type, and the usual `nan` checking (a couple of lines later) with `np.isnan` doesn't work. `as_float_array` from `sklearn` is appropriate for this situation in my opinion as `force_all_finite=False` takes care of these critical edge cases, and `None` is converted to `np.nan` and `inf` remains as `np.inf`. Since the default behavior of `IterativeImputer` is to specify `max_value` and `min_value` as `None`, this is an important case to take care of.

Reply:
```python
if np.isscalar(self._min_value):
    self._min_value = np.full(X.shape[1], self._min_value)
self._min_value = check_array(self._min_value, force_all_finite=False)
```

Reply:
IMO, we should not support `None` in arrays. Just numbers.

Reply:
I'd handle the None case first and then use broadcast_to. I think that would be most readable. I am not sure why we support None to be honest! I'd be happy to change the default to +/-inf in a different pr

Reply:
I agree to support None as a scalar. I would not introduce the support for array and the next PR could make a deprecation (actually IterativeImputer is still experimental) for None toward +/-inf which will make thing consistent between array and scalar.